runner/sidecar/source/db/db.go

```diff
@@ -0,0 +1,220 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	dfv1 "github.com/argoproj-labs/argo-dataflow/api/v1alpha1"
+	"github.com/argoproj-labs/argo-dataflow/runner/sidecar/source"
+	sharedutil "github.com/argoproj-labs/argo-dataflow/shared/util"
+	_ "github.com/go-sql-driver/mysql"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+var logger = sharedutil.NewLogger()
+
+const offsetTableSchema = `CREATE TABLE IF NOT EXISTS argo_dataflow_offsets (
+	table_name VARCHAR(255) NOT NULL,
```

Comment:
rather than table name, perhaps this could be uid, and uid is hash(cluster+namespace+pipeline+step+source)?

Reply:
That's the idea of another column `consumer_group`.

Reply:
I think we should allow people to specify queries that select from many tables, this is flexible, and will be better performing and much simpler that having to side-load tables. Hence my thinking that we should not have table name here.

Reply:
done.